src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -165,10 +192,33 @@ def run(self):
                     # Increase concurrency: create more threads.
                     log.debug("creating new threads to job increase")
                     for i in range(c, self.nr_concurrent_jobs):
-                        self.threads.append(JobThread(self, self.vc, name="{0}.{1}.{2}".format(self.name_pfx, time.time(), i)))
-                        self.threads[-1].start()
+                        if not self.run_event.is_set():
+                            break
+                        self.spawn_new_thread()
                 self.cv.wait(timeout=5)
 
+    def pause(self):
+        self.run_event.clear()
+
+        self.saved_q = deepcopy(self.q)
+        log.info('pause() saved jobs, cancelling jobs now...')
+        self.cancel_all_jobs()
+        log.info('pause() all jobs cancelled')
+        self.threads.clear()
```

Comment:
This is off-putting. Why do we need to clear the threads list? Once ongoing jobs are canceled, all threads should block (wait) on a condition variable (events in your case).

Reply:
Removed.